pandas/core/generic.py

```diff
@@ -8834,7 +8833,7 @@ def _where(
             # GH#21947 we have an empty DataFrame/Series, could be object-dtype
             cond = cond.astype(bool)
 
-        cond = -cond if inplace else cond
+        cond = ~cond if inplace else cond
```

Comment:
This was also breaking because of nullable booleans (the test `test_where_nullable_boolean_mask` from `pandas/tests/frame/indexing/test_where.py` in this PR fails).

Currently we have (which may be another bug):
```python
In [1]: import pandas as pd

In [2]: df = pd.DataFrame(pd.array([True, False]))

In [3]: -df
Out[3]:
    0
0  -1
1   0

In [4]: ~df
Out[4]:
       0
0  False
1   True
```

It might be because we're going through an object dtype array:

```python
In [10]: -np.array([True, False], dtype=object)

Out[10]: array([-1, 0], dtype=object)
```